zarr/convenience.py

```diff
@@ -348,3 +350,392 @@ def load(store):
     elif contains_group(store, path=None):
         grp = Group(store=store, path=None)
         return LazyLoader(grp)
+
+
+class _LogWriter(object):
```

Comment:
This is neat. I wonder if we should start using it with other things.

Also might be worth giving [this]( https://docs.python.org/3/howto/logging-cookbook.html#using-a-context-manager-for-selective-logging ) a look.